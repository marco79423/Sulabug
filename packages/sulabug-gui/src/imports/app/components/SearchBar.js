import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import Input from '@material-ui/core/Input'
import Button from '@material-ui/core/Button'
import {fromEvent} from 'rxjs'
import {filter} from 'rxjs/operators'

const styles = (theme) => createStyles({
  root: {
    width: '100%',
    display: 'flex',
  },
  inputWrapper: {
    flex: 1,
    background: theme.palette.background.paper,
  },
  input: {
    flex: 1,
    background: theme.palette.background.paper,
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
  },
  button: {
    marginLeft: theme.spacing.unit,
  },
})

class SearchBar extends React.Component {

  state = {
    searchTerm: '',
  }

  constructor(props) {
    super(props)

    this.inputRef = React.createRef()
  }

  searchComic = () => {
    this.props.searchComic(this.state.searchTerm)
  }

  handleChange = () => {
    this.setState({
      searchTerm: event.target.value
    })
  }

  componentDidMount() {
    fromEvent(this.inputRef.current, 'keyup').pipe(filter(event => event.key === 'Enter'))
      .subscribe(this.searchComic)
  }

  render() {
    const {classes} = this.props
    return (
      <div className={classes.root}>
        <Input
          inputRef={this.inputRef}
          autoFocus={true}
          placeholder="請輸入想試看的漫畫……"
          onChange={this.handleChange}
          className={classes.input}
          disableUnderline
          inputProps={{
            'aria-label': 'Description',
          }}
        />
        <Button variant="contained" size="small" color="secondary" className={classes.button}
                onClick={this.searchComic}>
          搜尋
        </Button>
      </div>
    )
  }
}

export default withStyles(styles)(SearchBar)
