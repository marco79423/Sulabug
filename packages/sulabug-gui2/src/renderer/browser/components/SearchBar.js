import React, {useEffect, useRef, useState} from 'react'
import {withRouter} from 'react-router-dom'
import {makeStyles, useTheme} from '@material-ui/core/styles'
import {fromEvent} from 'rxjs'
import {filter} from 'rxjs/operators'
import Input from '@material-ui/core/Input'
import Button from '@material-ui/core/Button'

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    display: 'flex',
  },
  inputWrapper: {
    flex: 1,
    background: theme.palette.background.paper,
  },
  input: {
    flex: 1,
    background: theme.palette.background.paper,
    paddingLeft: theme.spacing(1),
    paddingRight: theme.spacing(1),
  },
  button: {
    marginLeft: theme.spacing(1),
  },
}))

function SearchBar({history}) {
  const theme = useTheme()
  const classes = useStyles(theme)
  const inputRef = useRef()
  const [searchTerm, setSearchTerm] = useState('')

  useEffect(() => {
    fromEvent(inputRef.current, 'keyup').pipe(filter(event => event.key === 'Enter'))
      .subscribe(searchComic)
  }, [])

  const searchComic = () => {
    history.push(`/browser/browse?pattern=${searchTerm}`)
  }

  const handleChange = (e) => {
    setSearchTerm(e.target.value)
  }

  return (
    <div className={classes.root}>
      <Input
        inputRef={inputRef}
        autoFocus={true}
        placeholder="請輸入想試看的漫畫……"
        onChange={handleChange}
        className={classes.input}
        disableUnderline
        inputProps={{
          'aria-label': 'Description',
        }}
      />
      <Button variant="contained" size="small" color="secondary" className={classes.button}
              onClick={searchComic}>
        搜尋
      </Button>
    </div>
  )
}

export default withRouter(SearchBar)
