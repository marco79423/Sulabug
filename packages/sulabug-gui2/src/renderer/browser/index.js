import React, {useEffect} from 'react'
import {useDispatch} from 'react-redux'
import {Redirect, Route, Switch} from 'react-router-dom'
import {createMuiTheme, ThemeProvider} from '@material-ui/core/styles'
import {lime, teal} from '@material-ui/core/colors'

import CollectionPage from './pages/CollectionPage'
import BrowsePage from './pages/BrowsePage'
import SettingsPage from './pages/SettingsPage'
import DownloadPage from './pages/DownloadPage'


export default function Browser() {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch({type: 'common/initialize-app', data: 'browser'})
  }, [])

  const theme = createMuiTheme({
    typography: {
      h2: {
        fontSize: '1.6rem',
      },
      subtitle2: {
        fontSize: '0.8rem',
      },
      body2: {
        fontSize: '0.9rem',
      }
    },
    palette: {
      primary: {
        main: teal[800]
      },
      secondary: {
        main: lime[600]
      },
    },
  })

  return (
    <ThemeProvider theme={theme}>
      <Switch>
        <Redirect exact from='/browser' to='/browser/collection'/>
        <Route exact path='/browser/collection' component={CollectionPage}/>
        <Route exact path='/browser/browse' component={BrowsePage}/>
        <Route exact path='/browser/download' component={DownloadPage}/>
        <Route exact path='/browser/settings' component={SettingsPage}/>
      </Switch>
    </ThemeProvider>
  )
}
