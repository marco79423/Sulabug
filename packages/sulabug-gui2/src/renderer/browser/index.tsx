import React, {useEffect, useReducer} from 'react'
import {ipcRenderer} from 'electron'
import {Link} from 'react-router-dom'

console.log('browser module loaded')

function reducer(state, action) {
  switch (action.type) {
    case 'add-log':
      return {...state, logs: [...state.logs, action.data]}
    default:
      return state
  }
}

export default function Browser() {
  const [state, dispatch] = useReducer(reducer, {logs: []})
  useEffect(() => {
    ipcRenderer.on('sulabug-action', (e, action) => {
      dispatch(action)
    })
  }, [])

  return (
    <div>
      browser <Link to={`/manager`}>manager</Link>
      <ul>
        {state.logs.map((log, i) => <li key={i}>{log}</li>)}
      </ul>
    </div>
  )
}
