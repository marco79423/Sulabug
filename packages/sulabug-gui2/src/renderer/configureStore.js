import {applyMiddleware, combineReducers, compose, createStore} from 'redux'
import {forwardToMain, replayActionRenderer} from 'electron-redux'
import createSagaMiddleware from 'redux-saga'

import managerReducer from './manager/reducer'
import browserReducer from './browser/reducer'
import rootSaga from './rootSaga'


export default function configureStore() {
  const reducer = combineReducers({
    manager: managerReducer,
    browser: browserReducer,
  })

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose

  const sagaMiddleware = createSagaMiddleware()

  const store = createStore(
    reducer,
    composeEnhancers(applyMiddleware(
      forwardToMain, // IMPORTANT! This goes first
      sagaMiddleware,
    ))
  )

  sagaMiddleware.run(rootSaga)

  replayActionRenderer(store)

  return store
}
