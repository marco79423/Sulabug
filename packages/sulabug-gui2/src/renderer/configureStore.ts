import {applyMiddleware, combineReducers, compose, createStore} from 'redux'
import managerReducer from './manager/reducer'
import browserReducer from './browser/reducer'

declare global {
  interface Window { __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: any; }
}


export default function configureStore() {
  const reducer = combineReducers({
    manager: managerReducer,
    browser: browserReducer,
  })

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose

  const store = createStore(
    reducer,
    composeEnhancers(applyMiddleware(
    ))
  )

  return store
}
