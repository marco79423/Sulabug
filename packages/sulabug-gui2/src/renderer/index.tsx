import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
