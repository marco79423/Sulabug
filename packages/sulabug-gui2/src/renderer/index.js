import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
