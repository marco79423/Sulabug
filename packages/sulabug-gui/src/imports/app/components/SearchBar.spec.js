import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'

import SearchBar from './SearchBar'


describe('SearchBar', () => {
  it('will render search bar correctly', async () => {
    const component = renderer.create(
      <SearchBar
        searchComic={() => {
        }}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
