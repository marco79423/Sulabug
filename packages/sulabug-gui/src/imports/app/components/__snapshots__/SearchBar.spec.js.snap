// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar will render search bar correctly 1`] = `
<div
  className="SearchBar-root-1"
>
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 SearchBar-input-3"
    onClick={[Function]}
  >
    <input
      aria-label="Description"
      autoFocus={true}
      className="MuiInputBase-input-28 MuiInput-input-13"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="請輸入想試看的漫畫……"
      type="text"
    />
  </div>
  <button
    className="MuiButtonBase-root-61 MuiButton-root-35 MuiButton-contained-46 MuiButton-containedSecondary-48 MuiButton-raised-49 MuiButton-raisedSecondary-51 MuiButton-sizeSmall-58 SearchBar-button-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-36"
    >
      搜尋
    </span>
    <span
      className="MuiTouchRipple-root-64"
    />
  </button>
</div>
`;
